### PR TITLE
Timeseries: Fix outside range stale state

### PR DESCRIPTION
**Pull request by ryantxu**
This fixes an issue where the time range used to show a "helpful" link became out-of-sync with the expected state.  This caused the popup to be displayed even when the data is clearly within the time range.

### CODE REVIEW

File: public/app/plugins/panel/timeseries/plugins/OutsideRangePlugin.tsx [hunk 1]
```diff
@@ -1,34 +1,43 @@
...
   const plotInstance = useRef<uPlot>();
+  const [timevalues, setTimeValues] = useState<number[] | TypedArray>([]);
+  const [timeRange, setTimeRange] = useState<Scale | undefined>();
 
   useLayoutEffect(() => {
     config.addHook('init', (u) => {
       plotInstance.current = u;
     });
+
+    config.addHook('draw', (u) => {
```
Comment (toddtreece) [inline public/app/plugins/panel/timeseries/plugins/OutsideRangePlugin.tsx:22]: not sure if this is a bad idea, but `setScale` and `setData` didn't seem to work on this branch

Comment (leeoniya) [reply public/app/plugins/panel/timeseries/plugins/OutsideRangePlugin.tsx:22]: yes, draw hook fires on every resized pixel, so not good.

**Comment by leeoniya** [comment]
i'll poke at this tonight to see if anything can be improved

File: public/app/plugins/panel/timeseries/plugins/OutsideRangePlugin.tsx (hunk 1 above)
Comment (toddtreece) [reply public/app/plugins/panel/timeseries/plugins/OutsideRangePlugin.tsx]: i think i got it in [be94c48](https://github.com/grafana/grafana/pull/49633/commits/be94c486508da27e0f73b80a7d7b175df41e0b9f) using `setScale` by itself

**Review by toddtreece — APPROVED** [review]
the fix works for me locally

**Pull request merged**